### PR TITLE
Add IncludeXmlCommentsForAssembly convience overload

### LINKED ISSUE #2750: Add IncludeXmlComments overload that accepts assembly
Hey!

Thank you for this awesome project. 
Frequently I need to include Xml comments into Swagger, so I have to call `IncludeXmlComments` and pass here name of Xml file. It's inconvenient to hard code file name, and I have to construct it from assembly name. 
I propose to include following convenience helper/overload:

```csharp
    private static void IncludeXmlCommentsForAssembly(
        this SwaggerGenOptions swaggerGenOptions,
        Assembly assembly,
        bool includeControllerXmlComments = false)
        => swaggerGenOptions.IncludeXmlComments(
                 Path.Combine(AppContext.BaseDirectory, $"{assembly.GetName().Name}.xml"), 
                 includeControllerXmlComments);

```

If you like it, I'll submit a PR

### PR DESCRIPTION
**Pull request by leotsarev**
# Pull Request

## The issue or feature being addressed

Fixed #2750

### CODE REVIEW

File: test/Swashbuckle.AspNetCore.SwaggerGen.Test/XmlComments/XmlCommentsDocumentFilterTests.cs [hunk 1]
```diff
@@ -155,6 +154,25 @@ public void Ensure_IncludeXmlComments_Adds_Filter_To_Options()
...
+        [Fact]
+        public void Ensure_IncludeXmlCommentsForAssembly_Adds_Filter_To_Options()
+        {
+            var services = new ServiceCollection();
+            services.AddSingleton<IWebHostEnvironment, DummyHostEnvironment>();
+            services.AddSwaggerGen(c =>
+            {
+                c.IncludeXmlCommentsForAssembly(
+                    typeof(FakeControllerWithXmlComments).Assembly,
+                    includeControllerXmlComments: true);
```
Comment (martincostello) [inline test/Swashbuckle.AspNetCore.SwaggerGen.Test/XmlComments/XmlCommentsDocumentFilterTests.cs:166]: This makes me wonder if you could make this _even more_ convenient with an overload such as:

```csharp
// Not sure what the best name would be
public static void IncludeXmlComments<T>(
    this SwaggerGenOptions swaggerGenOptions,
    bool includeControllerXmlComments = false)
{
    var assembly = typeof(T).Assembly;
    return swaggerGenOptions.IncludeXmlCommentsForAssembly(assembly, includeControllerXmlComments);
}

// Usage
c.IncludeXmlComments<FakeControllerWithXmlComments>(includeControllerXmlComments: true);
```

**Comment by martincostello** [comment]
What about updating some of the tests and the README to use this? [code search](https://github.com/search?q=repo%3Adomaindrivendev%2FSwashbuckle.AspNetCore%20AppContext.BaseDirectory&type=code)

File: src/Swashbuckle.AspNetCore.SwaggerGen/DependencyInjection/SwaggerGenOptionsExtensions.cs [hunk 2]
```diff
@@ -556,6 +558,26 @@ public static void SupportNonNullableReferenceTypes(this SwaggerGenOptions swagg
...
+        /// Inject human-friendly descriptions for Operations, Parameters and Schemas based on XML comments
+        /// from specific Assembly
+        /// </summary>
+        /// <param name="swaggerGenOptions"></param>
+        /// <param name="assembly">Assembly that contains XML Comments</param>
+        /// <param name="includeControllerXmlComments">
+        /// Flag to indicate if controller XML comments (i.e. summary) should be used to assign Tag descriptions.
+        /// Don't set this flag if you're customizing the default tag for operations via TagActionsBy.
+        /// </param>
+        public static void IncludeXmlCommentsForAssembly(
```
Comment (martincostello) [inline src/Swashbuckle.AspNetCore.SwaggerGen/DependencyInjection/SwaggerGenOptionsExtensions.cs:571]: I'm not sure we need the `ForAssembly` prefix, as isn't that implied by the parameters and /// docs already?

File: test/Swashbuckle.AspNetCore.SwaggerGen.Test/XmlComments/XmlCommentsDocumentFilterTests.cs (hunk 1 above)
Comment (leotsarev) [reply test/Swashbuckle.AspNetCore.SwaggerGen.Test/XmlComments/XmlCommentsDocumentFilterTests.cs:166]: I think this is incorrect layering. We are operating in assembly terms, and while we are using class for locating assembly in our code, we don't actually perform any action on class. I prefer it be explicit in caller's code.

I'm investigated providing `Assembly.GetCallingAssembly()` overload, but it seems to be unstable

May be it worth to provide method that uses `GetEntryAssembly()`

File: src/Swashbuckle.AspNetCore.SwaggerGen/DependencyInjection/SwaggerGenOptionsExtensions.cs (hunk 2 above)
Comment (leotsarev) [reply src/Swashbuckle.AspNetCore.SwaggerGen/DependencyInjection/SwaggerGenOptionsExtensions.cs:571]: Ok, removed it

**Comment by leotsarev** [comment]
> What about updating some of the tests and the README to use this? [code search](https://github.com/search?q=repo%3Adomaindrivendev%2FSwashbuckle.AspNetCore%20AppContext.BaseDirectory&type=code)

Updated tests and README

File: test/Swashbuckle.AspNetCore.SwaggerGen.Test/XmlComments/XmlCommentsDocumentFilterTests.cs (hunk 1 above)
Comment (martincostello) [reply test/Swashbuckle.AspNetCore.SwaggerGen.Test/XmlComments/XmlCommentsDocumentFilterTests.cs]: The entry assembly is almost always wrong in tests, so I don't think we should introduce it.

File: README.md [hunk 3]
```diff
@@ -539,7 +539,8 @@ To enhance the generated docs with human-friendly descriptions, you can annotate
          );
 
          var filePath = Path.Combine(System.AppContext.BaseDirectory, "MyApi.xml");
-         c.IncludeXmlComments(filePath);
+         c.IncludeXmlComments(Assembly.GetExecutingAssembly());
+        // or c.IncludeXmlComments(typeof(MyController).Assembly));
```
Comment (martincostello) [inline README.md:543]:
```suggestion
         // or c.IncludeXmlComments(typeof(MyController).Assembly));
```

**Review by martincostello — APPROVED** [review]

File: src/Swashbuckle.AspNetCore.SwaggerGen/DependencyInjection/SwaggerGenOptionsExtensions.cs [hunk 4]
```diff
@@ -1,10 +1,12 @@
 using System;
 using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
 using System.Xml.XPath;
-using Microsoft.OpenApi.Models;
```
Comment (leotsarev) [inline src/Swashbuckle.AspNetCore.SwaggerGen/DependencyInjection/SwaggerGenOptionsExtensions.cs:4]: Automatically sorted usings. Should I revert this change?

Comment (martincostello) [reply src/Swashbuckle.AspNetCore.SwaggerGen/DependencyInjection/SwaggerGenOptionsExtensions.cs:4]: No, that's fine.

**Comment by leotsarev** [comment]
I don't understand why windows build is failing. Could we restart this to understand if this is some kind of instability?

**Comment by martincostello** [comment]
#2893

File: README.md [hunk 5]
```diff
@@ -539,7 +539,8 @@ To enhance the generated docs with human-friendly descriptions, you can annotate
          );
 
          var filePath = Path.Combine(System.AppContext.BaseDirectory, "MyApi.xml");
```
Comment (WeihanLi) [inline README.md:541]: maybe this line could be removed
